src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/DefaultRazorCompletionFactsService.cs

```diff
@@ -22,9 +22,15 @@ internal class DefaultRazorCompletionFactsService : RazorCompletionFactsService
             CSharpCodeParser.TagHelperPrefixDirectiveDescriptor,
         };
 
-        public override IReadOnlyList<RazorCompletionItem> GetCompletionItems(RazorSyntaxTree syntaxTree, SourceSpan location)
+        public override IReadOnlyList<RazorCompletionItem> GetCompletionItems(RazorCodeDocument codeDocument, SourceSpan location)
         {
+            if (codeDocument is null)
+            {
+                throw new ArgumentNullException(nameof(codeDocument));
+            }
+
             var completionItems = new List<RazorCompletionItem>();
+            var syntaxTree = codeDocument.GetSyntaxTree();
```

Comment:
Because of this change, we probably remove FileKind from ParserOptions. Can just use `codeDocument.GetFileKind()` here like everywhere else. Just a suggestion. Don't do it if it takes more than 5 minutes.

Reply:
Why is that better though? I get that we could. I don't get why we want *less* information available while parsing?

Reply:
Since i'm not passing in code document anymore can't do this anyhow 😄